crates/red_knot_python_semantic/resources/mdtest/metaclass.md

```diff
@@ -161,12 +162,16 @@ reveal_type(B.__class__)  # revealed: Literal[M]
 
 ## Non-class
 
-When a class has an explicit `metaclass` that is not a class, the value should be returned as is.
+When a class has an explicit `metaclass` that is not a class, but is a callable that accepts
+`type.__new__` arguments, we should return its return type.
 
 ```py
-class A(metaclass=1): ...
+def f(*args, **kwargs) -> int: ...
+
+class A(metaclass=f): ...
 
-reveal_type(A.__class__)  # revealed: Literal[1]
+# TODO should be `int`
```

Comment:
If the metaclass is a callable that returns the type `int`, that means the variable `A` created by the `class` statement will be an instance of the class `int`, which means `A.__class__` will be the `int` class itself, which means this revealed type should in fact be the meta-type of `int`, i.e...

```suggestion
# TODO should be `type[int]`
```

Reply:
Ah yeah, good catch!